TimberWinR/Outputs/Redis.cs

```diff
@@ -198,10 +228,16 @@ private void RedisSender()
                         try
                         {
                             string[] messages;
+                            // Exclusively
                             lock (_locker)
-                            {
-                                messages = _jsonQueue.Take(_batchCount).ToArray();
+                            {           
+                                // Take a sample of the queue depth
+                                if (_sampleCountIndex >= QUEUE_SAMPLE_SIZE)
```

Comment:
Might you choose to but this code and the UpdateCurrentBatchCount() call on line 302 closer together? How often does this loop run?